File: Sorts & Search & Shuffling Algorithms/Elementary Sorts/Comb Sort/Comb.py
class Comb:
    @staticmethod
    def getNextGap(gap):
        gap = (gap * 10) // 13
        if gap < 1:
            return 1
        return gap

    @staticmethod
    def combSort(arr):
        n = len(arr)
        gap = n
        swapped = True
        while gap != 1 or swapped == 1:
            gap = Comb.getNextGap(gap)
            swapped = False
            for i in range(0, n - gap):
                if arr[i] > arr[i + gap]:
                    arr[i], arr[i + gap] = arr[i + gap], arr[i]
                    swapped = True

File: Sorts & Search & Shuffling Algorithms/Elementary Sorts/Comb Sort/Comb.cpp
#include <vector>
#include <cstdlib>
#include <ctime>
using namespace std;
class Comb {
private:
    template<typename T>
    static void swap(T *xp, T *yp){
        T temp = *xp;
        *xp = *yp;
        *yp = temp;
    }
    static int getNextGap(int gap){
        gap = (gap*10)/13;

        if (gap < 1)
            return 1;
        return gap;
    }
public:
    template<typename T>
    static void Sort(vector<T> &it) {
        int n=it.size();
        int gap = n;
        bool swapped = true;
        while (gap != 1 || swapped == true){
            gap = getNextGap(gap);
            swapped = false;
            for (int i=0; i<n-gap; i++){
                if (it[i] > it[i + gap]){
                    swap(&it[i], &it[i + gap]);
                    swapped = true;
                }
            }
        }
    }
};